### PR TITLE
Draft: Add security handbook

### LINKED ISSUE #114: Update docs to describe authentication in federations
To accompany this merge, let's update the documentation here:

https://www.lf-lang.org/docs/handbook/target-declaration

Perhaps a new page in the Reference section of the handbook would be a good idea, rather than putting everything in the above file?  Then point to that page from the above?

The description in this PR would be an excellent starting point for that page.

_Originally posted by @edwardalee in https://github.com/lf-lang/lingua-franca/pull/1432#pullrequestreview-1208790472_

### PR DESCRIPTION
**Pull request by Jakio815**
This PR resolves #114, and is a handbook update of the security when federated execution. The recently merged PRs are lf-lang/reactor-c#105 and lf-lang/lingua-franca#1432 .

### CODE REVIEW

**Comment by Jakio815** [comment]
@lhstrh For github tests, I see "First-time contributors need a maintainer to approve running workflows." I would like some approval.

File: packages/documentation/copy/en/reference/Security.md
Comment (edwardalee) [inline packages/documentation/copy/en/reference/Security.md]:
```suggestion
The RTI build must include cmake options to enable simple HMAC-based authentication of federates. Add `-DAUTH=ON` option to the cmake command as shown below:
```

**Review by edwardalee — CHANGES_REQUESTED** [review]
Great to be updating this page! But the page as proposed is confusing. See comments.

File: packages/documentation/copy/en/reference/Security.md
Comment (Jakio815) [reply packages/documentation/copy/en/reference/Security.md]: @edwardalee Thanks for checking my PR. I fixed what you mentioned on the latest commit 7972f11 . Is there any other suggestions to add or change?

File: packages/documentation/copy/en/reference/Security.md
Comment (lhstrh) [inline packages/documentation/copy/en/reference/Security.md:9]:
```suggestion
By default, there is no secure authentication happening when a federate joins a federation, and data exchanged by federates is not encrypted. The `auth` target property can be used to change this.
```

File: packages/documentation/copy/en/reference/Security.md
Comment (lhstrh) [inline packages/documentation/copy/en/reference/Security.md:21]:
```suggestion
For the `C` target, federated execution is able to apply security with authentication by using HMAC authentication between RTI and federates. To enable this, include the `auth` property in your target specification, as follows:
```

File: packages/documentation/copy/en/reference/Security.md
Comment (lhstrh) [inline packages/documentation/copy/en/reference/Security.md:29]:
```suggestion
The RTI build must include CMake options to enable simple HMAC-based authentication of federates. Add `-DAUTH=ON` option to the CMake command as shown below:
```

File: packages/documentation/copy/en/reference/Security.md
Comment (lhstrh) [inline packages/documentation/copy/en/reference/Security.md:46]: It is unclear how this text relates to the target property. Isn't encryption implied by the use of the target property?

File: packages/lingua-franca/src/lib/documentationNavigation.ts
Comment (lhstrh) [inline packages/lingua-franca/src/lib/documentationNavigation.ts:229]:
```suggestion
          oneline: "Secure Federated Execution",
```

**Review by lhstrh — CHANGES_REQUESTED** [review]
Thanks, @Jakio815. I left some comments...

File: packages/documentation/copy/en/reference/Security.md
Comment (Jakio815) [reply packages/documentation/copy/en/reference/Security.md]: Commited all your changes at once including this and below.

File: packages/documentation/copy/en/reference/Security.md
Comment (lhstrh) [reply packages/documentation/copy/en/reference/Security.md:46]: What about this part? This text seems out of place...

Comment (Jakio815) [reply packages/documentation/copy/en/reference/Security.md]: I added more details of this part. The encryption is not applied to the messages for communication yet, and this part is the description of the plan.

Comment (lhstrh) [reply packages/documentation/copy/en/reference/Security.md]: Hm, this sounds like a very experimental feature then. Should we even be documenting it if it is so incomplete? I suggest we put a fat warning message at the very beginning of the page about this so that users don't enable the feature with the expectation their federation will be secure...

Comment (Jakio815) [reply packages/documentation/copy/en/reference/Security.md]: Yeah. I was actually starting to doubt myself 'Does the handbook need to show future works.' 
Then I'll erase the future work part and add a warning message.

Comment (lhstrh) [reply packages/documentation/copy/en/reference/Security.md]: I think we can safely leave out the details for future work. Those would be good to document in an issue, however.

File: packages/documentation/copy/en/reference/Security.md
Comment (lhstrh) [inline packages/documentation/copy/en/reference/Security.md:9]:
```suggestion
### Warning: the experimental security features described on this page are under development and not production ready. Users should not expect their federations will be secure if the `auth` target property is enabled.
```

File: packages/documentation/copy/en/reference/Security.md
Comment (lhstrh) [inline packages/documentation/copy/en/reference/Security.md:11]:
```suggestion
By default, there is no secure authentication happening when a federate joins a federation, and data exchanged by federates is not encrypted. For targets that support it, the `auth` target property can be used to enable authentication between federates. Messages exchanged between federates after authentication are **not encrypted**, but this capability is planned for the future.
```

**Review by lhstrh — CHANGES_REQUESTED** [review]
I've made a couple more suggested edits. I think that it makes little sense to have authentication if there is no encryption afterwards, however, so I think we should only include this in the manual if encryption indeed will be added in the near future.

**Comment by Jakio815** [comment]
> I've made a couple more suggested edits. I think that it makes little sense to have authentication if there is no encryption afterwards, however, so I think we should only include this in the manual if encryption indeed will be added in the near future.

Thanks for the edits. I committed it, so please don't mind those resolved comments. It was just automatically resolved by Git.

What do you mean include it in the manual? Are you not meaning this handbook?

**Comment by lhstrh** [comment]
> What do you mean include it in the manual? Are you not meaning this handbook?

Yes, that's what I meant.

**Review by lhstrh — APPROVED** [review]

**Comment by lhstrh** [comment]
> I think we should only include this in the manual if encryption indeed will be added in the near future.

To clarify, you are planning to add this, right? If so, let's merge.

**Comment by Jakio815** [comment]
> > I think we should only include this in the manual if encryption indeed will be added in the near future.
> 
> To clarify, you are planning to add this, right? If so, let's merge.

Yes it's true. I was going to merge it but, I see that I have to get approval from Prof. Lee. I think it's because if there are 'changes requested' I have to get a review.

**Pull request merged**
